src/test/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMBuilderTest.java

```diff
@@ -1044,8 +1044,8 @@ public void given__server_branch_norev_userpass__when__build__then__scmBuilt() t
     public void given__server_branch_norev_userkey__when__build__then__scmBuilt() throws Exception {
         createGitHubSCMSourceForTest(true, "https://github.test/tester/test-repo.git");
         BranchSCMHead head = new BranchSCMHead("test-branch");
-        source.setCredentialsId("user-key");
         GitHubSCMBuilder instance = new GitHubSCMBuilder(source, head, null);
+        instance.withCredentials("user-key", GitHubSCMBuilder.SSH);
```

Comment:
Nit: prefer non deprecated API usage
```suggestion
        GitHubSCMBuilder instance = new GitHubSCMBuilder(source, head, revision)
                .withCredentials("user-key")
                .withResolver(GitHubSCMBuilder.SSH);
```